### PR TITLE
upstreams: initial boilerplate/skeleton for metadata upstream

### PR DESCRIPTION
**Pull request by tbarrella**
Commit Message:
upstreams: initial boilerplate/skeleton for metadata upstream

Signed-off-by: Taylor Barrella <tabarr@google.com>

Additional Description:

A small commit so that it's easier to review. This is meant to be the smallest testable unit, establishing the code location and some naming

cc @lambdai

Risk Level: Low
Testing: Unit
Docs Changes: N/A
Release Notes: N/A
#15472

### CODE REVIEW

File: source/extensions/upstreams/http/metadata/BUILD [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "//bazel:envoy_build_system.bzl",
+    "envoy_cc_library",
+    "envoy_extension_package",
+)
+
+licenses(["notice"])  # Apache 2
+
+envoy_extension_package()
+
+envoy_cc_library(
```
Comment (lambdai) [inline source/extensions/upstreams/http/metadata/BUILD:11]: Please use `envoy_cc_extension`
and add `category` and `security_posture`

See: https://github.com/envoyproxy/envoy/blob/main/source/extensions/upstreams/http/BUILD#L11

File: test/extensions/upstreams/http/metadata/BUILD [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "//bazel:envoy_build_system.bzl",
+    "envoy_cc_test",
+    "envoy_package",
+)
+
+licenses(["notice"])  # Apache 2
+
+envoy_package()
+
+envoy_cc_test(
```
Comment (lambdai) [inline test/extensions/upstreams/http/metadata/BUILD:11]: envoy_extension_cc_test

**Review by lambdai — CHANGES_REQUESTED** [review]
LGTM modulo build target and comments

**Comment by lambdai** [comment]
Also: I am not sure if you should add to https://github.com/envoyproxy/envoy/blob/main/source/extensions/extensions_build_config.bzl at this moment

File: test/extensions/upstreams/http/metadata/BUILD (hunk 2 above)
Comment (tbarrella) [reply test/extensions/upstreams/http/metadata/BUILD:12]: I added a TODO for this; it requires adding to `extensions_build_config.bzl` which requires a config registration class

**Pull request closed without merging**